utHelper.py

```diff
@@ -370,3 +370,124 @@ def saveLogForAnalysis(self, inputLog, filename):
             # Write each line of the log data to the file
             for writeString in inputLog:
                 fileHandle.write(writeString + "\n")
+
+    def downloadFileToDevice(self, url, target_directory):
+        """
+        Download the file and copy to device directory.
+
+        Args:
+            url (str): url path.
+            target_directory (str): target directory on device.
+        """
+        self.log.step("url( url:'{}' )".format(url))
+        self.log.step("target_directory( target_directory:'{}' )".format(target_directory))
+
+        working_directory = self.outboundClient.workspaceDirectory
+        file_name = os.path.basename(url)
+        self.outboundClient.downloadFile(url, file_name)
+        self.copyFileFromHost(os.path.join(working_directory, file_name), target_directory)
```

Comment:
need to loop round... Not sure why you need to be adding the working directory here. Since outboundClient downloads into the working directory?